### PR TITLE
Pull in changes from CppXPort unit tests

### PR DESCRIPTION
**Pull request by seowwj**

### CODE REVIEW

**Review by hongdami — CHANGES_REQUESTED** [review]
transformer.h: should use templete for the datatype. this can wait for a later time
transformer.cc: the filename should be put into cmake file. I tried to build, Error appeared.

**Review by hongdami — CHANGES_REQUESTED** [review]
Try to run testcase and you may find:
undefined symbol: _ZTIN22message_infrastructure19IdentityTransformerE

**Review by hongdami — CHANGES_REQUESTED** [review]
return_type = self.mp.build_actor(target_fn, builder, idx=1)

function args ...
We should run the testcase pass

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/message_infrastructure_py_wrapper.cc [hunk 1]
```diff
@@ -161,6 +161,11 @@ PYBIND11_MODULE(MessageInfrastructurePywrapper, m) {
     .def("service", &CppVarPortScalarSparse::Service)
     .def("recv", &CppVarPortScalarSparse::Recv)
     .def("peek", &CppVarPortScalarSparse::Peek);
+  py::class_<IdentityTransformer> (m, "IdentityTransformer")
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/message_infrastructure_py_wrapper.cc:164]: We need to change the transformer logic. The python side transfer a function to achieve the transformer.

File: src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py [hunk 2]
```diff
@@ -43,6 +46,88 @@ def target_fn(*args, **kwargs):
...
+        """
+        Spawns an actor and kills it after a certain time.
+        Checks that multiprocessing daemon will automatically restart crashed
+        actor and reassign to correct process.
+        """
+        self.test_multiprocessing_spawn()
+
+        # Gets list of actor PIDs and kills the 1st one in the list
+        actor_list = self.mp.actor_pids
+        os.kill(actor_list[0])
```
Comment (hongdami) [inline src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py]: we kill the actor not from python side.

Comment (seowwj) [reply src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py]: for unit testing, shouldn't we simulate a scenario where the actor "randomly" crashes / dies, to see if the module can automatically restart the actor?

Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py]: for actor "randomly" crashes/dies, I move the restart work to actor in the latest commit. The multiprocessor will send a signal to actor for stopping slowly

Comment (seowwj) [reply src/lava/magma/runtime/message_infrastructure/test/test_multiprocessing.py]: Can you share with me which file has been changed? In the latest commit (https://github.com/hongdami/lava/commit/b71b5cf8f0a211c88478186e392aedb881bae10e) it seems that only the PyWrapper has been changed

**Pull request merged**